cmake/onnxruntime_providers_cuda.cmake

```diff
@@ -158,6 +158,13 @@
       set(onnxruntime_NVCC_THREADS "1" CACHE STRING "Number of threads that NVCC can use for compilation.")
       target_compile_options(${target} PRIVATE "$<$<COMPILE_LANGUAGE:CUDA>:SHELL:--threads \"${onnxruntime_NVCC_THREADS}\">")
     endif()
+    
+    # relocatable-device-code=true 
+    if (MSVC AND CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 12.8)
+      set_target_properties(${target} PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
```

Comment:
Please try separate compilation with link time optimization:
https://developer.nvidia.com/blog/improving-gpu-app-performance-with-cuda-11-2-device-lto/
That might have better performance than separate compilation.

Reply:
I think this requires performance testing as some report perf loss and overhead of the feature.

Reply:
So far, this PR shows 1.2% perf regression on bert-large and no regression on onnx zoo models on EP_Perf CI. Will enable LTO and test perf diff on https://github.com/microsoft/onnxruntime/pull/23671